samza-api/src/main/java/org/apache/samza/table/descriptors/BaseTableDescriptor.java

```diff
@@ -78,33 +68,33 @@ public String getTableId() {
     return tableId;
   }
 
-  /**
-   * Get the serde assigned to this {@link TableDescriptor}
-   *
-   * @return {@link KVSerde} used by this table
-   */
-  public KVSerde<K, V> getSerde() {
-    return serde;
+  @Override
+  public Map<String, String> toConfig(Config jobConfig) {
+
+    validate();
+
+    Map<String, String> tableConfig = new HashMap<>(config);
+    tableConfig.put(
+        String.format(JavaTableConfig.TABLE_PROVIDER_FACTORY, tableId),
+        getProviderFactoryClassName());
+    generateConfig(jobConfig, tableConfig);
```

Comment:
The pattern we're using in StreamDescriptors is to have the extending classes also override the toConfig method from the base class and call `super.toConfig` as the first step. Might be a simpler interface for extending classes since there is one less config related interface method. If we do that, you might also not need to call addTableConfig to mutate the passed in `tableConfig` map in generateConfig.